post.js

```diff
@@ -54,48 +54,70 @@
     return copied;
   }
 
-  stbvorbis.decode = (buf) => {
-    return new Promise(async (resolve, reject) => {
-      await initializationP;
-      let copiedBuf = null;
-      if (buf instanceof ArrayBuffer) {
-        copiedBuf = arrayBufferToHeap(buf, 0, buf.byteLength);
-      } else if (buf instanceof TypedArray) {
-        copiedBuf = arrayBufferToHeap(buf.buffer, buf.byteOffset, buf.byteLength);
-      }
-      const channelsPtr = Module._malloc(4);
-      const sampleRatePtr = Module._malloc(4);
-      const outputPtr = Module._malloc(4);
-      const length = decodeMemory(copiedBuf.byteOffset, copiedBuf.byteLength, channelsPtr, sampleRatePtr, outputPtr);
-      if (length < 0) {
-        reject(new Error('stbvorbis decode failed: ' + length));
-        return;
-      }
-      const channels = ptrToInt32(channelsPtr);
-      
-      let data = [];
-      const dataPtrs = ptrToInt32s(ptrToInt32(outputPtr), channels);
-      for (const ptr of dataPtrs) {
-        data.push(ptrToFloat32s(ptr, length));
-      }
-      const result = {
-        data:       data,
-        sampleRate: ptrToInt32(sampleRatePtr),
-      };
+  onmessage = async event => {
+    await initializationP;
+    const buf = event.data.arrayBuffer;
+    let copiedBuf = null;
+    if (buf instanceof ArrayBuffer) {
+      copiedBuf = arrayBufferToHeap(buf, 0, buf.byteLength);
+    } else if (buf instanceof TypedArray) {
```

Comment:
Your previous change was reverted...?

(I think it is ok to replace this in another PR.)

Reply:
This is an obvious bug, so I'll fix this soon.

Reply:
Done. Please resolve the conflict.

Reply:
it was not intented revert. 😢